core/arch/instr_shared.c

```diff
@@ -1299,7 +1299,11 @@ instr_expand(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr)
         !instr_valid(instr))
         return instr;
 
-    DOLOG(5, LOG_ALL, { loginst(dcontext, 4, instr, "instr_expand"); });
+    DOLOG(5, LOG_ALL, {
+        instr_t *log_instr = instr_clone(dcontext, instr);
+        loginst(dcontext, 4, log_instr, "instr_expand");
```

Comment:
I don't understand this one: how does cloning the instr change anything?  This needs a comment explaining, as it is unusual.

Reply:
The LOG changes the instr (makes operands valid). I don't think we should do that (LOG shouldn't influence the behavior) => I cloned it to make the LOG have its own instr to print.

Reply:
Re-open if not ok

Reply:
I see, that makes sense, just not obvious, so a comment will help.

Reply:
Just so it's clear, I've been requesting that a comment be added *to the code*, not just this conversation in the review.

Reply:
Done